lib/model/model.go

```diff
@@ -1465,12 +1465,10 @@ func sendIndexTo(minSequence int64, conn protocol.Connection, folder string, fs
 func (m *Model) updateLocalsFromScanning(folder string, fs []protocol.FileInfo) {
 	m.updateLocals(folder, fs)
 
-	// Fire the LocalChangeDetected event to notify listeners about local
-	// updates.
+	// Fire the LocalChangeDetected event to notify listeners about local updates.
 	m.fmut.RLock()
-	path := m.folderCfgs[folder].Path()
+	m.localChangeDetected(m.folderCfgs[folder], fs)
 	m.fmut.RUnlock()
-	m.localChangeDetected(folder, path, fs)
```

Comment:
Before I was calling the lock on the 'folderConfiguration.Path()' method and immediately releasing.  Now I am passing in the whole folder object 'm.folderCfgs[folder]' so I can get a few more things from it.  I was originally going to run the lock again only on that part but then Jakob wrote 

> The locking is needed where you access m.folderCfgs, as that is a non-concurrent-safe map that may be changed by other routines in parallell.

So I translated that to mean... lock everything that touches this until it stops touching it.